magento/magento2
# Add price rule attributes when hydrating product during price calcula…

…tion

### Description
Column in `catalog_eav_attribute` is `is_used_for_price_rules`. This PR means that all attributes used for price calculation are fetched by the `lowestPriceOptionsProvider()` (which is called inside `module-configurable-product/Pricing/Price/ConfigurablePriceResolver.php`->`resolvePrice`) 

Backstory / justification: 

I was adding a feature to control price visibility (for some products, we don't want to display the price on our magento store). This includes for some options of configurable products.  Despite setting `is_used_for_price_rules` to be true, the product model is hydrated with only some default attributes (special prices, tax class), but we need to be able to add some extra processing to check we're allowed to display to price


### Manual testing scenarios
1. Create a new attribute, for example, `price_visibility`, and ensure that it's corresponding entry in `catalog_eav_attribute` has `is_used_for_price_rules` to be true
2. Add this attribute to a configurable product's attribute set, and set some of the simple products to have this as false
3. Create a new configurable price resolver and attempt to filter sub products so that only those with price_visibility of true are considered. 

### Contribution checklist
